File: frontend/src/api/models/bibliography.ts
export type Person = {
  firstName: string;
  lastName: string;
  middleInitial: string;
}

export type CitationRecord = {
  id: string;
  title: string;
  authors: string;
  volNum: string;
  date: string;
  startYear: string;
  endYear: string;
  pages: string;
  doi: string;
  author: Person;
  contributers: string;
  version: string;
  number: string;
  publicationLocale: string;
  format: string;
  accessDate: string;
  publisher: string;
  type: string;
  chapter: string;
  editors: string;
  translators: string;
  fullString: string;
}

export type Bibliography = {
  id: string;
  label: string;
  citations: Array<CitationRecord>
}

export type UserBibliographies = {
  id: string;
  uid: string;
  bibliographies: Array<Bibliography>
}

File: middle/models/bibliography.py
from pydoc import doc
from pydantic import BaseModel
from typing import List,Dict

class Person(BaseModel):
    firstName: str
    lastName: str
    middleInitial: str

class Citation(BaseModel):
    title: str
    authors: str
    volNum: str
    date: str
    startYear: str
    endYear: str
    pages: str
    doi: str
    author: Person
    contributors: str
    version: str
    number: str
    publicationlocale: str
    format: str
    accessDate: str
    publisher: str
    type: str
    chapter: str
    editors: str
    translators: str
    fullString: str


class Bibliography(BaseModel):
    label: str
    citations: List[Citation]

class UserBibliographies(BaseModel):
    uid: str
    bibliographies: List[Bibliography]